plugins/node/instrumentation-fs/src/instrumentation.ts

```diff
@@ -0,0 +1,321 @@
+          if (hasPromises) {
+            for (const fName of PROMISE_FUNCTIONS) {
+              if (isWrapped(fs.promises[fName])) {
+                this._unwrap(fs.promises, fName);
+              }
+              this._wrap(
+                fs.promises,
+                fName,
+                <any>this._patchPromiseFunction.bind(this, fName)
+              );
+            }
+          }
+          for (const fName of ASYNC_FUNCTIONS) {
+            if (isWrapped(fs[fName])) {
+              this._unwrap(fs, fName);
+            }
+            this._wrap(
+              fs,
+              fName,
+              <any>this._patchAsyncFunction.bind(this, fName)
+            );
+          }
+          for (const fName of SYNC_FUNCTIONS) {
+            if (isWrapped(fs[fName])) {
+              this._unwrap(fs, fName);
+            }
+            this._wrap(
+              fs,
+              fName,
+              <any>this._patchSyncFunction.bind(this, fName)
+            );
+          }
+          return fs;
+        },
+        (fs: FS) => {
+          if (fs === undefined) return;
+          this._diag.debug('Removing patch for fs');
+          if (hasPromises) {
+            for (const fName of PROMISE_FUNCTIONS) {
+              if (isWrapped(fs.promises[fName])) {
+                this._unwrap(fs.promises, fName);
+              }
+            }
+          }
+          for (const fName of ASYNC_FUNCTIONS) {
+            if (isWrapped(fs[fName])) {
+              this._unwrap(fs, fName);
+            }
+          }
+          for (const fName of SYNC_FUNCTIONS) {
+            if (isWrapped(fs[fName])) {
+              this._unwrap(fs, fName);
+            }
+          }
+        }
+      ),
+    ];
+  }
+
+  protected _patchPromiseFunction<T extends (...args: any[]) => ReturnType<T>>(
+    fnName: string,
+    original: T
+  ): T {
+    const instrumentation = this;
+    return <any>async function (this: any, ...args: any[]) {
+      if (isTracingSuppressed(api.context.active())) {
+        // Performance optimization. Avoid creating additional contexts and spans
+        // if we already know that the tracing is being suppressed.
+        return original.apply(this, args);
+      }
+      const { createHook, endHook } =
+        instrumentation.getConfig() as FsInstrumentationConfig;
+      if (typeof createHook === 'function') {
+        if (
+          // promise and async variants get mixed here for the hooks
+          createHook(fnName, {
+            args: args,
+          }) === false
+        ) {
+          // return original.apply(this, args);
+          return api.context.with(
+            suppressTracing(api.context.active()),
+            original,
+            this,
+            ...args
+          );
+        }
+      }
+
+      const span = instrumentation.tracer.startSpan(`fs ${fnName}`) as api.Span;
+      try {
+        // QUESTION: Should we immediately suppress all internal nested calls?
```

Comment:
not sure which nested calls you are refering to ?

Reply:
`fs.writeFile` for example calls `open` and `close` internally. Latter are the nested calls I'm referring to.

Reply:
Personally I would. Maybe we can introduce a verbose mode later if users ask for it

Reply:
I'm going to let you make a decision here as i can see it both ways. As I said I would lean towards suppressing and we can always unsuppress or add a verbose mode later. Hard to go back the other way without breaking alerts and stuff.